contracts/simple_dex/lib.rs

```diff
@@ -373,14 +373,27 @@ mod simple_dex {
             let this = self.env().account_id();
             let balance_token_in = self.balance_of(token_in, this)?;
             let balance_token_out = self.balance_of(token_out, this)?;
-
             if min_amount_token_out.map_or(false, |x| balance_token_out < x) {
                 // throw early if we cannot support this swap anyway due to liquidity being too low
                 return Err(DexError::NotEnoughLiquidityOf(token_out));
             }
 
+            Self::_out_given_in(
+                amount_token_in,
+                balance_token_in,
+                balance_token_out,
+                self.swap_fee_percentage,
+            )
+        }
+
+        fn _out_given_in(
+            amount_token_in: Balance,
+            balance_token_in: Balance,
+            balance_token_out: Balance,
+            swap_fee_percentage: Balance,
+        ) -> Result<Balance, DexError> {
```

Comment:
I wonder - can it be the case that the returned `Balance` here (the amount of tokens the caller will be paid out in exchange for his `amount_token_in`) is less than `min_amount_token_out` he requested? We check only whether `balance_token_out > min_amount_token_in` but can it be the case that our formula `_out_given_in` will give less than that in the end? After the swap fee, etc.

Reply:
@fbielejec pointed me to https://github.com/Cardinal-Cryptography/aleph-node/blob/7e5d524caa933dae6331465d146d1e40b86ccb7f/contracts/simple_dex/lib.rs#L173 which I totally missed.